src/lib/components/ConfirmTransactions.js

```diff
@@ -12,78 +17,201 @@ ${json}
+
+  const widgetSrc = props.widgetSrc;
+
+  const getWidgetContractPermission = async (widgetSrc, contractId) =>
+    await cache.asyncLocalStorageGet(StorageDomain, {
+      widgetSrc,
+      contractId,
+      type: StorageType.SendTransactionWithoutConfirmation,
+    });
+
+  useEffect(() => {
+    (async () => {
+      if (transactions.length === 1) {
+        const contractId = transactions[0].contractName;
+        const isSignedIntoContract = await near.isSignedIntoContract(contractId);
+
+        const widgetContractPermission = await getWidgetContractPermission(widgetSrc, contractId);
+
+        const dontAskForConfirmation = (isSignedIntoContract && widgetContractPermission && widgetContractPermission[transactions[0].methodName])
+          ? true : false;
+
+        setDontAskForConfirmation(dontAskForConfirmation);
+
+        if (dontAskForConfirmation) {
+          setLoading(true);
+          const result = await near.sendTransactions(transactions);
+          setLoading(false);
+          onHide(result);
+        }
+      } else {
+        setDontAskForConfirmation(false);
+      }
+    })();
+  }, []);
+
   const onHide = props.onHide;
-  const transactions = props.transactions;
+
   const show = !!transactions;
 
-  return (
-    <Modal size="xl" centered scrollable show={show} onHide={onHide}>
-      <Modal.Header closeButton>
-        <Modal.Title>Confirm Transaction</Modal.Title>
-      </Modal.Header>
-      <Modal.Body>
-        {transactions &&
-          transactions.map((transaction, i) => (
-            <div key={`${gkey}-${i}`}>
-              <div>
-                <h4>Transaction #{i + 1}</h4>
-              </div>
-              <div>
-                <span className="text-secondary">Contract ID: </span>
-                <span className="font-monospace">
-                  {transaction.contractName}
-                </span>
-              </div>
-              <div>
-                <span className="text-secondary">Method name: </span>
-                <span className="font-monospace">{transaction.methodName}</span>
-              </div>
-              {transaction.deposit && transaction.deposit.gt(0) && (
+  const dontAskAgainCheckboxChange = async () => {
+    setDontAskAgainChecked(!dontAskAgainChecked);
+    setDontAskAgainErrorMessage(null);
+  };
+
+  if (dontAskForConfirmation) {
+    const transaction = transactions[0];
+    return (
+      <ToastContainer position="bottom-end" className="position-fixed">
+        <Toast show={show} bg="info">
+          <Toast.Header>
+            Sending transaction {Loading}
+          </Toast.Header>
+          <Toast.Body>
+            Calling contract <span className="font-monospace">{transaction.contractName}</span> with method <span className="font-monospace">{transaction.methodName}</span>
+          </Toast.Body>
+        </Toast>
+      </ToastContainer>
+    );
+  } else if (dontAskForConfirmation === false) {
```

Comment:
As I mentioned before (https://github.com/NearSocial/VM/pull/156#discussion_r1425768505), please, refactor it so it is easier to maintain this code in the future as to me `if (x) {} else if (x === false) {} else {}` is really hard to reason about and I imagine future developers will make wrong assumptions about this code. Replace it with:

```ts
if (x === null) {
  return ...;
}
if (x) {
} else {
}
```

Reply:
Good point. I've moved the check for `null` to the top now.

I would also suggest introducing a test framework to avoid future regressions, but maybe we want a separate PR for that?

Reply:
yeah, regression testing should be implemented separately

Reply:
From another point of view. If there will still be some time before this will be merged, I could add tests so that we have "proof" of the scenarios covered here. It's always reassuring to deliver a PR with tests to prove that what you deliver works as it should. Let me know what you think.